modules/storage/cosmos-db/main.bicep

```diff
@@ -1,99 +1,353 @@
-@description('Deployment Location')
-param location string
+@description('Deployment region name. Default is the location of the resource group.')
+param location string = resourceGroup().location
 
-@description('Prefix of Cosmos DB Resource Name')
-param prefix string = enableCassandra ? 'coscas' : 'cosmos'
+@description('The bakend API type of Cosmos DB database account. The API selection cannot be changed after account creation. Possible values: "cassandra", "gremlin", "mongodb", "sql", "table".')
+@allowed([ 'cassandra', 'gremlin', 'mongodb', 'sql', 'table' ])
+param backendApi string = 'cassandra'
 
-@description('Name of Cosmos DB Resource')
-param name string = '${prefix}${uniqueString(resourceGroup().id, location)}'
+@description('The name of the Cosmos DB account. Character limit: 3-44, valid characters: lowercase letters, numbers, and hyphens. It must me unique across Azure.')
+param name string = 'cdb${backendApi}-${uniqueString(resourceGroup().id, location)}'
 
-@description('Max stale requests. Required for BoundedStaleness. Valid ranges, Single Region: 10 to 2147483647. Multi Region: 100000 to 2147483647.')
+@description('Enables automatic failover of the write region in the rare event that the region is unavailable due to an outage. Automatic failover will result in a new write region for the account and is chosen based on the failover priorities configured for the account.')
+param enableAutomaticFailover bool = true
+
+@description('Multi-region writes capability allows you to take advantage of the provisioned throughput for your databases and containers across the globe.')
+param enableMultipleWriteLocations bool = true
+
+@description('Enable Serverless for consumption-based usage.')
+param enableServerless bool = false
+
+@description('Flag to indicate whether or not this region is an AvailabilityZone region')
+param isZoneRedundant bool = false
```

Comment:
Very good thing to add. Calling this out as a reminder to myself to include in guidelines.